### ISSUE TITLE
Web APIs: representations of non-information resources

### ISSUE THREAD

**Issue by alik-kirillovich** [RubenVerborgh/WebFundamentals#89]
The _Web APIs_ lecture [says](https://rubenverborgh.github.io/WebFundamentals/web-apis/#resources) «Non-information resources cannot be represented as bits: people, phenomena, concepts, ideas…».

(This statement concerns REST resources, but it is relevant to web resources too, since [the slide](https://rubenverborgh.github.io/WebFundamentals/web-apis/#web-resources) on web resources refers to it).

However, it is not clear, whether there are standards that explicitly prove this statement.

_[Architecture of the World Wide Web](https://www.w3.org/TR/webarch/)_ just defines an information resource as «a resource which has the property that all of its essential characteristics can be conveyed in a message». This definition suggests what a non-information resource can be, but doesn't clarify the question of whether it can be represented.

It seems, that the truth of the statement in question depends on what the nature of a resource representation is. There are two possible options:

1. A resource representation is some form of information _about_ this resource. In this case, both information and non-information resources can have representations. For example, a photo of the Eiffel Tower in JPEG format can be a representation of non-information resource Eiffel Tower. The only difference is that representation of an information resource can also be the value of this resource.

2. A resource representation is some form of information that is _a value_ of this resource, that constitutes the resource's _essence_, that the resource _embodies_ or _consists of_. In this case, of course, only information resources can have representations.

Again, I can't find standards that explicitly confirms any of these two above-mentioned options:

- _[Architecture of the World Wide Web](https://www.w3.org/TR/webarch/)_ defines representation as «data that encodes information about resource state», and doesn't provide any details about what this information can be. This definition is perfectly consistent with option N.1, but it doesn't contradict option N.2 too.

- _[[httpRange-14] Resolved](https://lists.w3.org/Archives/Public/www-tag/2005Jun/0039.html)_ only states, that an assumed representation of a non-information resource (if any) can't be directly obtained via a GET request with a 2xx response.

- _[Cool URIs for the Semantic Web](https://www.w3.org/TR/cooluris/)_ notes, that we should distinguish between resource representations and resource descriptions. But it doesn't clarify what a resource representation is, and how it differed from a resource description.

- _[Dereferencing HTTP URIs ](https://www.w3.org/2001/tag/doc/httpRange-14/2007-08-31/HttpRange-14.html)_ draft implicitly endorses option N.2, claiming that the Wikipedia entry, RDF triples and so on can't be representations of the non-information resource planet Mars, because these representation can't convey the essence of the planet. This explanation depends on an implicit presupposition that a resource representation must convey the essence of this resource. However, this presupposition itself has been never asserted in this draft. The draft only says that a resource representation is «a form of the information, associated with a resource, that can be transmitted across the Web», and never says that it must convey the essence. So, this endorsement of option N.2 seems groundless.

It would be nice, if this issue can be somehow explained on these slides.

**Comment by RubenVerborgh** [comment]
Thanks, just two small comments:

> Architecture of the World Wide Web just defines a non-information resource as «a resource which has the property that all of its essential characteristics can be conveyed in a message» and doesn't address the question of its possible representation.

(Typo: that should be "information resources".) Note that "conveyed in a message" and "being able to be represented" are equivalent to me.

> A resource representation is some form of information about this resource.

With HTTP Range 14, I would say that the "about" document is a different resource altogether.

**Comment by alik-kirillovich** [comment]
> Typo: that should be "information resources".

I reformulated the paragraph to fix this typo.

**Comment by alik-kirillovich** [comment]
>Note that "conveyed in a message" and "being able to be represented" are equivalent to me.

Indeed, this equivalence is presupposed by the statement in question. However it is not so obvious taken in isolation. In colloquial language and in semantics, «representation» means some kind of a relation between a sign and a real-world object. A portrait is a representation of a person, a map is a representation of a territory and a scientific theory is a representation of natural phenomena. In this sense, representations can be of both information and non-information resources. Of course, in the context of web resources, the term «representation» can have a special technical meaning. In fact, the main point of the raised issue is that it would be nice, if the slides explicitly clarify the difference between the colloquial and the technical meanings, as well as provide standards (if any) where this technical meaning is defined.
